src/resources/item/item.model.ts

```diff
@@ -2,54 +2,60 @@ import mongoose, {Document, Schema} from 'mongoose'
 import Item from './item.interface'
 export interface ItemDocument extends Document, Item {}
 
-const opts = {toJSON: {virtuals: true}}
+const itemSchema = new Schema({
+	name: {
+		type: String,
+		required: true,
+	},
+	note: {
+		type: String,
+	},
+	quantity: {
+		type: Number,
+		min: 1,
+		default: 1,
+		required: true,
+	},
+	price: {
+		type: Number,
+		min: 0,
+		required: true,
+	},
+	total: {
+		type: Number,
+		required: true,
+		default: 0,
+	},
+	date: {
+		type: Date,
+		default: Date.now(),
+		required: true,
+	},
+	category: {
+		required: true,
+		type: Schema.Types.ObjectId,
+		ref: 'category',
+	},
+	team: {
+		required: true,
+		type: Schema.Types.ObjectId,
+		ref: 'team',
+	},
+	creator: {
+		required: true,
+		type: Schema.Types.ObjectId,
+		ref: 'user',
+	},
+})
+
+itemSchema.pre<ItemDocument>('save', function(next) {
+	if (!this.isModified('price') || !this.isModified('quantity')) {
```

Comment:
I think both price and quantity is required in this model. So if each of them does not exist, mongoose wouldn't even allow it to come here?

Reply:
True. But this one I check to see if it is modified or not. If we update item but not modify the price or quantiry then the total is not recalculated